docs/dev/table/sqlClient.md

```diff
@@ -0,0 +1,539 @@
+{% endhighlight %}
+
+{% top %}
+
+### Environment Files
+
+A SQL query needs a configuration environment in which it is executed. The so-called *environment files* define available table sources and sinks, external catalogs, user-defined functions, and other properties required for execution and deployment.
+
+Every environment file is a regular [YAML file](http://http://yaml.org/) that looks similar to the following example. The file defines an environment with a table source `MyTableName` that reads from CSV file. Queries that are executed in this environment (among others) will have a parallelism of 1, an even-time characteristic, and will run in the `table` result mode.
+
+{% highlight yaml %}
+# Define table sources and sinks here.
+
+tables:
+  - name: MyTableName
+    type: source
+    schema:
+      - name: MyField1
+        type: INT
+      - name: MyField2
+        type: VARCHAR
+    connector:
+      type: filesystem
+      path: "/path/to/something.csv"
+    format:
+      type: csv
+      fields:
+        - name: MyField1
+          type: INT
+        - name: MyField2
+          type: VARCHAR
+      line-delimiter: "\n"
+      comment-prefix: "#"
+
+# Execution properties allow for changing the behavior of a table program.
+
+execution:
+  type: streaming
+  time-characteristic: event-time
+  parallelism: 1
+  max-parallelism: 16
+  min-idle-state-retention: 0
+  max-idle-state-retention: 0
+  result-mode: table
+
+# Deployment properties allow for describing the cluster to which table programs are submitted to.
+
+deployment:
+  response-timeout: 5000
+{% endhighlight %}
+
+Environment files can be created for general purposes (*defaults environment file* using `--defaults`) as well as on a per-session basis (*session environment file* using `--environment`). Every CLI session is initialized with the default properties followed by the session properties. Both default and session environment files can be passed when starting the CLI application. If no default environment file has been specified, the SQL Client searches for `./conf/sql-client-defaults.yaml` in Flink's configuration directory.
+
+Properties that have been set within a CLI session (e.g. using the `SET` command) have highest precedence:
+
+{% highlight text %}
+CLI commands > session environment file > defaults environment file
+{% endhighlight %}
+
+{% top %}
+
+### Dependencies
+
+The SQL Client does not require to setup a Java project using Maven or SBT. Instead, you can pass the dependencies as regular JAR files that get submitted to the cluster. You can either specify each JAR file separately (using `--jar`) or define entire library directories (using `--library`). For connectors to external systems (such as Apache Kafka) and corresponding data formats (such as JSON), Flink provides **ready-to-use JAR bundles**. These JAR files are suffixed with `sql-jar` and can be downloaded for each release from the Maven central repository.
+
+{% if site.is_stable %}
```

Comment:
It's kinda weird to have `-sql-jar` in previous section but not showing these section when `site.is_stable == false`, should we also include the previous paragraph in this if statement as well?